Sources/Spezi/Utilities/AnyOptional.swift

```diff
@@ -0,0 +1,31 @@
+//
+// This source file is part of the Stanford Spezi open-source project.
+// It is based on the code from the Apodini (https://github.com/Apodini/Apodini) projects.
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md) and the Apodini project authors
+//
+// SPDX-License-Identifier: MIT
+//
+
+/// A type erased `Optional`.
+///
+/// This is useful to unwrapping, e.g.,  generics or associated types by declaring an extension under the condition of
+/// ``AnyOptional`` conformance. This allows in the implementation of the extension to access the underlying
+/// ``Wrapped`` type of an `Optional`, essentially unwrapping the optional type.
+public protocol AnyOptional {
```

Comment:
Haha, remember this code and some of the decisions that went into writing it, good old times 😅

Reply:
Yes. I had the gut feeling that I had written that code before haha.
I tried several approaches to avoid having those protocols implemented, but seems like the most sensible way to me.
I thought I just make them public?

Reply:
Sure, that works for me, could be useful or some other use cases as well 👍